src/checks.jl

```diff
@@ -146,3 +146,132 @@ function EMB.check_node(n::HydroStorage, 𝒯, modeltype::EMB.EnergyModel, check
         "The field `level_min` can not be larger than 1."
     )
 end
+
+"""
+    EMB.check_node(n::HydroReservoir, 𝒯, modeltype::EMB.EnergyModel, check_timeprofiles::Bool)
```

Comment:
All checks are currently a bit limited, while the checks for the `AbstractConstraintType`s are repeated in each. These should be replaced with dispatching on [`check_node_data`](https://github.com/EnergyModelsX/EnergyModelsBase.jl/blob/adfb7a41d399192ab1e81b7e7fd50f695406217c/src/checks.jl#L78).

Reply:
I can implement the skeleton for `check_node_data`, but you two would need to add specific checks.

Reply:
Prepared for in https://github.com/EnergyModelsX/EnergyModelsRenewableProducers.jl/pull/35/commits/677e47200e02f26b5e939778dfb351c50b691a24
This implies we still need the improved checks.

Reply:
I've compared with the check that are implemented for the old `HydroStorage` and added to additional checks. Ref 82944d045d61fa36dd579adf67f19cb034c9781b

Reply:
Included additional check for constraints: non-negative penalty and valid resource type. 396c38fbe0b55f0764705f6919636f5289e293df